RecoMET/METAlgorithms/src/GlobalHaloAlgo.cc

```diff
@@ -231,5 +235,342 @@ reco::GlobalHaloData GlobalHaloAlgo::Calculate(const CaloGeometry& TheCaloGeomet
+      
+      ECALBmatched |= ebmatched;
+      ECALEmatched |= eematched;
+      HCALBmatched |= hbmatched;
+      HCALEmatched |= hematched;
+      
+    }
+  }
+
+
+  TheGlobalHaloData.SetSegmentIsEBCaloMatched(ECALBmatched);
+  TheGlobalHaloData.SetSegmentIsEECaloMatched(ECALEmatched);
+  TheGlobalHaloData.SetSegmentIsHBCaloMatched(HCALBmatched);
+  TheGlobalHaloData.SetSegmentIsHECaloMatched(HCALEmatched);
+  
+
+  //Now checking patterns from EcalHaloData and HcalHaloData: 
+  //Simply check whether any cluster has a halo pattern
+  //In that case store the rhits in GlobalHaloData 
+
+  bool HaloPatternFoundInEB =false;
+  for(auto & hcand : hccandEB){
+    if((hcand.getIsHaloFromPattern() &&!ishlt)||(hcand.getIsHaloFromPattern_HLT() &&ishlt)){
+      HaloPatternFoundInEB =true;
+      edm::RefVector<EcalRecHitCollection> bhrhcandidates = hcand.getBeamHaloRecHitsCandidates();
+      AddtoBeamHaloEBEERechits(bhrhcandidates, TheGlobalHaloData,true);
+    } 
+  }
+  
+
+  bool HaloPatternFoundInEE =false;
+  for(auto & hcand : hccandEE){
+    if((hcand.getIsHaloFromPattern() &&!ishlt)||(hcand.getIsHaloFromPattern_HLT() &&ishlt)){
+      HaloPatternFoundInEE =true;
+      edm::RefVector<EcalRecHitCollection> bhrhcandidates = hcand.getBeamHaloRecHitsCandidates();
+      AddtoBeamHaloEBEERechits( bhrhcandidates, TheGlobalHaloData,false);
+    } 
+  }
+
+
+  bool HaloPatternFoundInHB =false;
+  for(auto & hcand : hccandHB){
+    if((hcand.getIsHaloFromPattern() &&!ishlt)||(hcand.getIsHaloFromPattern_HLT() &&ishlt)){
+      HaloPatternFoundInHB =true;
+      edm::RefVector<HBHERecHitCollection> bhrhcandidates = hcand.getBeamHaloRecHitsCandidates();
+      AddtoBeamHaloHBHERechits(bhrhcandidates, TheGlobalHaloData);
+    } 
+  }
+  
+
+  bool HaloPatternFoundInHE =false;
+  for(auto & hcand : hccandHE){
+    if((hcand.getIsHaloFromPattern() &&!ishlt)||(hcand.getIsHaloFromPattern_HLT() &&ishlt)){
+      HaloPatternFoundInHE =true;
+      edm::RefVector<HBHERecHitCollection> bhrhcandidates = hcand.getBeamHaloRecHitsCandidates();
+      AddtoBeamHaloHBHERechits(bhrhcandidates, TheGlobalHaloData);
+    } 
+  }
+  TheGlobalHaloData.SetHaloPatternFoundEB(HaloPatternFoundInEB);
+  TheGlobalHaloData.SetHaloPatternFoundEE(HaloPatternFoundInEE);
+  TheGlobalHaloData.SetHaloPatternFoundHB(HaloPatternFoundInHB);
+  TheGlobalHaloData.SetHaloPatternFoundHE(HaloPatternFoundInHE);
+    if(
+     HaloPatternFoundInEB||HaloPatternFoundInEE||HaloPatternFoundInHB||HaloPatternFoundInHE||
+     ECALBmatched||ECALEmatched||HCALBmatched||HCALEmatched
+     ){
+  cout <<"CSCCaloMatched in EB/EE/HB/HE : "
+       <<ECALBmatched <<", " 
+       <<ECALEmatched <<", " 
+       <<HCALBmatched <<", " 
+       <<HCALEmatched <<endl;
+
+  cout << "HaloPatternFoundIn EB/EE/HB/HE : "
+       << HaloPatternFoundInEB<<", "
+       << HaloPatternFoundInEE<<", "
+       << HaloPatternFoundInHB<<", "
+       << HaloPatternFoundInHE<<", "<<endl;}
```

Comment:
these cout s have to go or be made LogDebug or LogTrace (will not be compiled by default, will need ```EDM_ML_DEBUG ``` and MessageLogger